File: app/models/preorder.rb
require 'autoinc'

class Preorder
  include Mongoid::Document
  include Mongoid::Timestamps
  include Mongoid::Autoinc

  field                     :user_vk_id
  field                     :comments
  field                     :status, default: 'Создан'
  field                     :number, type: Integer

  belongs_to                :drink
  belongs_to                :user
  has_and_belongs_to_many   :syurups

  increments :number, model_name: 'Preorder'
  index number: 1

  validates_uniqueness_of :number
end

File: frontend/tmp/babel-output_path-r3ZBTNX3.tmp/frontend/models/preorder.js
import DS from 'ember-data';
var Preorder;

Preorder = DS.Model.extend({
  comments: DS.attr('string'),
  status: DS.attr('string'),
  drink: DS.belongsTo('drink', {
    async: true
  }),
  user: DS.belongsTo('user', {
    async: true
  }),
  syurups: DS.hasMany('syurup', {
    async: true
  })
});

export default Preorder;